bin/charm-env

```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
+if [[ "$1" == "--version" || "$1" == "-v" ]]; then
+    echo 1.1.0
```

Comment:
I went with 1.1.0 because it's backwards compatible (so not 2.x) but it's also not 1.0.0 since it has features that older versions are missing.  Perhaps the default when version is not supported should be 1.0.0 instead of 0.0.0?

I also embedded the version in `charm-env` because that's the only thing it affects.  Changes to the rest of the layer shouldn't change the version of `charm-env` and perhaps vice versa.  I actually considered making `charm-env` into a snap, but that seemed like overkill.

Reply:
I moved it out into a global `VERSION` variable at the top of the file and moved the logic bit down a bit to make it more obvious and in your face.  I think that a VERSION file would actually be more likely to be missed, so I'm for leaving it embedded.